src/test/scala/com/microsoft/hyperspace/index/rules/FilterIndexRuleTest.scala

```diff
@@ -146,35 +150,56 @@ class FilterIndexRuleTest extends HyperspaceSuite {
     assert(transformedPlan.equals(originalPlan), "Plan should not transform.")
   }
 
-  private def verifyTransformedPlan(logicalPlan: LogicalPlan): Unit = {
-    logicalPlan match {
-      case Project(
-          _,
-          Filter(
+  test("Verify FilterIndex rule is applied when all columns are selected.") {
+    val filterCondition = And(IsNotNull(c4), EqualTo(c4, Literal(10, IntegerType)))
+    val originalPlan = Filter(filterCondition, scanNode)
+
+    val transformedPlan = FilterIndexRule(originalPlan)
+    assert(!transformedPlan.equals(originalPlan), "No plan transformation.")
+    verifyTransformedPlanWithIndex(transformedPlan, indexName2)
+  }
+
+  private def verifyTransformedPlanWithIndex(
+      logicalPlan: LogicalPlan,
+      indexName: String): Unit = {
+    val relation = logicalPlan.collect {
+      case l: LogicalRelation => l
+    }
+    assert(relation.length == 1)
+    relation.head match {
+      case l @ LogicalRelation(
+            HadoopFsRelation(
+              newLocation: InMemoryFileIndex,
+              newPartitionSchema: StructType,
+              dataSchema: StructType,
+              bucketSpec: Option[BucketSpec],
+              _: ParquetFileFormat,
+              _),
+            _,
             _,
-            l @ LogicalRelation(
-              HadoopFsRelation(
-                newLocation: InMemoryFileIndex,
-                newPartitionSchema: StructType,
-                dataSchema: StructType,
-                bucketSpec: Option[BucketSpec],
-                _: ParquetFileFormat,
-                _),
-              _,
-              _,
-              _))) =>
-        val allIndexes = IndexCollectionManager(spark).getIndexes(Seq(Constants.States.ACTIVE))
-        val expectedLocation = getIndexDataFilesPath(indexName)
-        assert(newLocation.rootPaths.head.equals(expectedLocation), "Invalid location.")
-        assert(newPartitionSchema.equals(new StructType()), "Invalid partition schema.")
-        assert(dataSchema.equals(allIndexes.head.schema), "Invalid schema.")
-        assert(bucketSpec.isEmpty, "Invalid bucket spec.")
+            _) =>
+        verifyIndexProperties(indexName, newLocation, newPartitionSchema, dataSchema, bucketSpec)
         assert(dataSchema.fieldNames.toSet.equals(l.output.map(_.name).toSet))
-
       case _ => fail("Unexpected plan.")
     }
   }
 
+  private def verifyIndexProperties(
+      indexName: String,
+      newLocation: InMemoryFileIndex,
+      newPartitionSchema: StructType,
+      dataSchema: StructType,
+      bucketSpec: Option[BucketSpec]): Unit = {
+    val allIndexes = IndexCollectionManager(spark).getIndexes(Seq(Constants.States.ACTIVE))
+    val expectedLocation = getIndexDataFilesPath(indexName)
+    assert(newLocation.rootPaths.head.equals(expectedLocation), "Invalid location.")
```

Comment:
This is an existing code, but can we remove the assert message "Invalid blah" in this block? I don't think it adds any value and it is just more burdensome for the developer.

Reply:
Done.